execution/commitment/hex_patricia_hashed.go

```diff
@@ -1322,6 +1463,13 @@ func (hph *HexPatriciaHashed) toWitnessTrie(hashedKey []byte, codeReads map[comm
 					fmt.Printf("[witness, pos %d] FullNodeChild Hash (%d, %0x, depth=%d) %s proof %+v\n", keyPos, row, col, hph.depths[row], currentCell.FullString(), fullNode.Children[col])
 				}
 			}
+
+			// this deals with the edge case where the extension key in nextNode diverges from hashedKey
+			// and points to a branch node. In this case we just need to provide the hash of this branch node
+			if pathDivergenceFound {
+				terminalCell := &hph.grid[row][currentNibble]
+				nextNode.(*trie.ShortNode).Val = trie.NewHashNode(terminalCell.hash[:])
```

Comment:
lets introduce special functions for those specific cases eg:
`.createDivergingPathNode(hph.grid[row][nibble])`